doc/dev/tests.md

```diff
@@ -0,0 +1,378 @@
+    def test_client_creation(self, schemaregistry_endpoint):
+        client = self.create_schemareg_client(schemaregistry_endpoint)
+        assert client is not None
+
+```
+
+There's a lot going on in the example so we'll take this piece by piece:
+
+* Import everything you will need in your tests as normal, add to your imports the line `from devtools_testutils import AzureTestCase, PowerShellPreparer`. These two objects give our tests a lot of the desired powers.
+* `AzureTestCase`: the test class should inherit from this object (`class TestSchemaRegistry(AzureTestCase)`), doing so sets up the recording infrastructure and the client creation methods.
+* `PowerShellPreparer`: this preparer serves two purposes.
+    * First, it will provide the live keys we need to test our library against live resources.
+    * Second, it will keep those same live keys out of our recordings to make sure that we are not leaking our secrets into the recordings.
+* At the top of your test class you should include any helper methods you will need. Most libraries will have a client creation method to eliminate repetitive code.
+* Following your helper methods will be your actual tests. All test methods must start with "test". The preparer built at the top of the file should decorate your test in the fashion: `@MyPreparer()`.
+    * The signature of your test will always contain `self`, and following self will be all the keys that you need from your preparer. A test does not need to have every key passed into it, the test framework will take care of passing in only the parameters specifically requested in the test signature.
+
+If you need logging functionality for your testing, pytest also offers [logging](https://docs.pytest.org/en/stable/logging.html) capabilities either inline through the `caplog` fixture or with command line flags.
+
+## An example test
+An example test for schemaregistry looks like:
+```python
+class AppConfigTestCase(AzureTestCase):
+
+    ...
+    @SchemaRegistryPreparer()
+    def test_schema_basic(self, schemaregistry_endpoint, schemaregistry_group):
+        client = self.create_client(schemaregistry_endpoint)
+        schema_name = self.get_resource_name('test-schema-basic')
+        schema_str = """{"namespace":"example.avro","type":"record","name":"User","fields":[{"name":"name","type":"string"},{"name":"favorite_number","type":["int","null"]},{"name":"favorite_color","type":["string","null"]}]}"""
+        serialization_type = "Avro"
+        schema_properties = client.register_schema(schemaregistry_group, schema_name, serialization_type, schema_str)
+
+        assert schema_properties.schema_id is not None
+        assert schema_properties.location is not None
+        assert schema_properties.location_by_id is not None
+        assert schema_properties.version is 1
+        assert schema_properties.serialization_type == "Avro"
+
+        with pytest.raises(HttpResponseError):
+            client.get_schema('a' * 32)
```

Comment:
Just add a comment for clarity "this is expected to fail since this doesn't exist"